06-classification_continued.Rmd

```diff
@@ -559,16 +657,378 @@ The overall workflow for performing K-nearest neighbour classification using `ti
+Figure \@ref(fig:06-fixed-irrelevant-features) corroborates
+this evidence; if we fix the number of neighbours to $K=3$, the accuracy falls off more quickly.
+
+```{r 06-neighbours-irrelevant-features, echo = FALSE, warning = FALSE, fig.retina = 2, out.width = "600", fig.cap = "Tuned number of neighbours for varying number of irrelevant predictors"}
+plt_irrelevant_nghbrs <- ggplot(res) +
+              geom_line(mapping = aes(x=ks, y=nghbrs)) +
+              labs(x = "Number of Irrelevant Predictors", y = "Number of neighbours")
+plt_irrelevant_nghbrs
+```
+
+```{r 06-fixed-irrelevant-features, echo = FALSE, warning = FALSE, fig.retina = 2, out.width = "600", fig.cap = "Accuracy versus number of irrelevant predictors for tuned and untuned number of neighbours"}
+res_tmp <- res %>% pivot_longer(cols=c("accs", "fixedaccs"), names_to="Type", values_to="accuracy")
+plt_irrelevant_nghbrs <- ggplot(res_tmp) +
+              geom_line(mapping = aes(x=ks, y=accuracy, color=Type)) +
+              labs(x = "Number of Irrelevant Predictors", y = "Accuracy") + 
+              scale_color_discrete(labels= c("Tuned K", "K = 3"))
+plt_irrelevant_nghbrs
+```
+
+**Finding a good subset of predictors**
+
+So then if we can't just use all of our variables as predictors without consideration, how 
+do we choose which variables we *should* use?  A simple method is to rely on your scientific understanding
+of the data to tell you which variables are not likely to be useful predictors. For example, in the cancer
+data that we have been studying, the `ID` variable is just a unique identifier for the observation.
+As it is not related to any measured property of the cells, the `ID` variable should therefore not be used
+as a predictor. That is, of course, a very clear-cut case. But the decision for the remaining variables 
+is less obvious, as all seem like reasonable candidates. It 
+is not clear which subset of them will create the best classifier. One could use visualizations and
+other exploratory analyses to try to help understand which variables are potentially relevant, but
+this process is both time-consuming and error-prone when there are many variables to consider.
+We therefore need a more systematic and programmatic way of choosing variables. 
+This is a very difficult problem to solve in
+general, and there are a number of methods that have been developed that apply
+in particular cases of interest; see the additional resources at the end of
+this chapter to find out where you can learn more about variable selection.
+
+The first idea you might think of for a systematic way to select predictors
+is *best subset selection*. Best subset selection essentially involves trying
+all possible subsets of variables and picking the "best" one. In particular, you
+
+1. create a separate model for every possible subset of predictors
+2. tune each one using cross validation
+3. pick the subset of predictors that gives you the highest estimated prediction accuracy  
```

Comment:
should we make this clear that this is cross-validated prediction accuracy?

Reply:
done